### PR TITLE
feat(ec2): Prefixlist Constructs

### PR DESCRIPTION
**Pull request by watany-dev**
I created this Construct so that I can use AWS::EC2::PrefixList quickly.

The differences from L1 are as follows:

- If AddressFamily is not specified, ipv4 is used by default.
- If maxEntries is not specified, the number of entries is automatically calculated from the entries array and set as maxEntries.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-ec2/lib/prefix-list.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+      entries: props.entries || [],
+    });
+
+    this.prefixListId = prefixList.attrPrefixListId;
+    this.arn = prefixList.attrArn;
+    this.ownerId = prefixList.attrOwnerId;
+    this.version = prefixList.attrVersion;
+    this.addressFamily = prefixList.addressFamily;
+
+    this.node.defaultChild = prefixList;
```
Comment (WinterYukky) [inline packages/@aws-cdk/aws-ec2/lib/prefix-list.ts:163]: The ID of the resource that will be the defaultChild should be 'Resource' and do not set the defaultChild directly as in `this.node.defaultChild = prefixList;`.

File: packages/@aws-cdk/aws-ec2/lib/prefix-list.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    if (props?.prefixListName) {
+      if ( props.prefixListName.startsWith('com.amazonaws')) {
+        throw new Error('The name cannot start with \'com.amazonaws.\'');
+      };
+      if (props.prefixListName.length > 255 ) {
+        throw new Error('Lengths exceeding 255 characters cannot be set.');
+      };
+    };
+
+    this.name = props?.prefixListName || id;
```
Comment (WinterYukky) [inline packages/@aws-cdk/aws-ec2/lib/prefix-list.ts:144]: `this.prefixListName = this.physicalName;`

File: packages/@aws-cdk/aws-ec2/test/prefix-list.test.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    }).toThrow('The name cannot start with \'com.amazonaws.\'');
+  });
+
+  test('invalid prefixlist-name over 255 characters', () => {
+    // GIVEN
+    const stack = new Stack();
+    expect(() => {
+      new PrefixList(stack, 'prefix-list', {
+        maxEntries: 100,
+        prefixListName: 'ThesupercalifragilisticexpialidociouswordisusedtodescribesomethingthatisextraordinarilygoodItwaspopularizedinthemovieMaryPoppinswhichwasreleasedin1964ThewordismadeupofacombinationofLatinandGreekrootsandithassincebecomeapartofpopularcultureWhileitmaybe difficulttospellitisafunwordtosayandcanbeusedtoimpressyourfriendsSothenexttimeyouwanttodescribesomethingasamazingtryusingthewordsupercalifragilisticexpialidocious',
```
Comment (WinterYukky) [inline packages/@aws-cdk/aws-ec2/test/prefix-list.test.ts:73]: It is difficult to see that the given `prefixListName` is more than 256 characters, so try the following.

```ts
new PrefixList(stack, 'prefix-list', {
        maxEntries: 100,
        prefixListName: 'x'.repeat(256),
});
```

File: packages/@aws-cdk/aws-ec2/README.md [hunk 4]
```diff
@@ -1703,3 +1703,24 @@ new ec2.Instance(this, 'Instance1', {
   detailedMonitoring: true,
 });
 ```
+
+## Managed prefix lists
+
+The following demonstrates how to enable [Detailed Monitoring](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-cloudwatch-new.html) for an EC2 instance. Keep in mind that Detailed Monitoring results in [additional charges](http://aws.amazon.com/cloudwatch/pricing/).
```
Comment (WinterYukky) [inline packages/@aws-cdk/aws-ec2/README.md:1709]: Why is a description of detailed monitoring included?

**Review by WinterYukky — CHANGES_REQUESTED** [review]
Thank you for opening the PR. I am commenting on this as I needed to brush up on a few things.

Please note that it is currently recommended to go through the RFC to create new L2 constructs in aws-cdk, and this PR may be closed to go through the recommended flow. For more information, please see the following document
https://github.com/aws/aws-cdk/blob/main/docs/NEW_CONSTRUCTS_GUIDE.md

File: packages/@aws-cdk/aws-ec2/README.md (hunk 4 above)
Comment (watany-dev) [reply packages/@aws-cdk/aws-ec2/README.md:1753]: I'm sorry. This is the "original description". :)

**Comment by watany-dev** [comment]
Thank you for the great comments that helped to resolve the issue. I have made some minor corrections. I will address the remaining comments later.

File: packages/@aws-cdk/aws-ec2/README.md [hunk 5]
```diff
@@ -1703,3 +1703,24 @@ new ec2.Instance(this, 'Instance1', {
...
+
+```ts
+new ec2.PrefixList(stack, 'empty-prefix-list', {
```
[the remark this answers is not in the document]
Comment (watany-dev) [reply packages/@aws-cdk/aws-ec2/README.md]: fixed.

File: packages/@aws-cdk/aws-ec2/test/prefix-list.test.ts (hunk 3 above)
Comment (watany-dev) [reply packages/@aws-cdk/aws-ec2/test/prefix-list.test.ts]: fixed.

File: packages/@aws-cdk/aws-ec2/lib/prefix-list.ts (hunk 2 above)
Comment (watany-dev) [reply packages/@aws-cdk/aws-ec2/lib/prefix-list.ts]: fixed.

File: packages/@aws-cdk/aws-ec2/lib/prefix-list.ts [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+   * The Name of the Prefix List
+   *
+   */
+  public readonly name: string;
+
+  /**
+   * The ARN of the Prefix List
+   *
+   */
+  public readonly arn: string;
```
[the remark this answers is not in the document]
Comment (watany-dev) [reply packages/@aws-cdk/aws-ec2/lib/prefix-list.ts]: fixed.

File: packages/@aws-cdk/aws-ec2/lib/prefix-list.ts [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+   *
+   * @attribute
+   */
+  public readonly prefixListId: string;
+
+  /**
+   * The Name of the Prefix List
+   *
+   */
+  public readonly name: string;
```
[the remark this answers is not in the document]
Comment (watany-dev) [reply packages/@aws-cdk/aws-ec2/lib/prefix-list.ts]: fixed.

File: packages/@aws-cdk/aws-ec2/lib/prefix-list.ts [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+      if (props.prefixListName.length > 255 ) {
+        throw new Error('Lengths exceeding 255 characters cannot be set.');
+      };
+    };
+
+    this.name = props?.prefixListName || id;
+
+    if (!props?.maxEntries && !props?.entries) {
+      throw new Error('Set maxEntries or enrties.');
+    }
```
[the remark this answers is not in the document]
Comment (watany-dev) [reply packages/@aws-cdk/aws-ec2/lib/prefix-list.ts]: fixed.

File: packages/@aws-cdk/aws-ec2/lib/prefix-list.ts (hunk 1 above)
Comment (watany-dev) [reply packages/@aws-cdk/aws-ec2/lib/prefix-list.ts]: fixed.

File: packages/@aws-cdk/aws-ec2/lib/prefix-list.ts [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+
+  /**
+   * The name of the prefix list.
+   *
+   * @default None
+   *
+   * @remarks
+   * It is not recommended to use an explicit name.
+   */
+  readonly prefixListName?: string;
```
[the remark this answers is not in the document]
Comment (watany-dev) [reply packages/@aws-cdk/aws-ec2/lib/prefix-list.ts:56]: thankyou! fixed.

File: packages/@aws-cdk/aws-ec2/lib/prefix-list.ts
[the remark this answers is not in the document]
Comment (watany-dev) [reply packages/@aws-cdk/aws-ec2/lib/prefix-list.ts:1]: Thank you. I have fixed it as much as I could.

**Pull request closed without merging**